File: app/src/main/java/org/michaelbel/moviemade/presentation/features/movie/BackdropDialog.java
package org.michaelbel.moviemade.presentation.features.movie;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import org.michaelbel.moviemade.presentation.App;
import org.michaelbel.moviemade.R;
import org.michaelbel.moviemade.presentation.common.bottomsheet.BottomSheetDialogFragment;
import org.michaelbel.moviemade.core.utils.IntentsKt;
import org.michaelbel.moviemade.core.utils.SharedPrefsKt;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatTextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;

public class BackdropDialog extends BottomSheetDialogFragment {

    private String path;
    private Unbinder unbinder;

    @Inject
    SharedPreferences preferences;

    @BindView(R.id.set_btn) AppCompatTextView setBtn;
    @BindView(R.id.cancel_btn) AppCompatTextView cancelBtn;

    static BackdropDialog newInstance(String path) {
        Bundle args = new Bundle();
        args.putString(IntentsKt.BACKDROP, path);
        BackdropDialog fragment = new BackdropDialog();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public int getTheme() {
        return R.style.BottomSheetTheme;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ((App) getActivity().getApplication()).createFragmentComponent().inject(this);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.dialog_backdrop, container, false);
        unbinder = ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        Bundle args = getArguments();
        if (args != null) {
            path = args.getString(IntentsKt.BACKDROP);
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        unbinder.unbind();
    }

    @OnClick(R.id.set_btn)
    public void setBtnClick(View v) {
        preferences.edit().putString(SharedPrefsKt.KEY_ACCOUNT_BACKDROP, path).apply();
        Toast.makeText(getActivity(), R.string.msg_image_applied, Toast.LENGTH_SHORT).show();
        dismiss();
    }

    @OnClick(R.id.cancel_btn)
    public void cancelBtnClick(View v) {
        dismiss();
    }
}

File: app/src/main/java/org/michaelbel/moviemade/presentation/features/movie/dialog/BackdropDialog.kt
package org.michaelbel.moviemade.presentation.features.movie.dialog

import android.content.SharedPreferences
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import kotlinx.android.synthetic.main.dialog_backdrop.*
import org.michaelbel.moviemade.R
import org.michaelbel.moviemade.core.local.SharedPrefs.KEY_ACCOUNT_BACKDROP
import org.michaelbel.moviemade.presentation.App
import javax.inject.Inject

class BackdropDialog: BottomSheetDialogFragment() {

    companion object {
        const val BACKDROP = "backdrop"

        internal fun newInstance(path: String): BackdropDialog {
            val args = Bundle()
            args.putString(BACKDROP, path)
            val fragment = BackdropDialog()
            fragment.arguments = args
            return fragment
        }
    }

    private var path: String? = null

    @Inject
    lateinit var preferences: SharedPreferences

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        App[requireActivity().application as App].createFragmentComponent().inject(this)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
        inflater.inflate(R.layout.dialog_backdrop, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        path = arguments?.getString(BACKDROP) ?: ""

        setBtn.setOnClickListener {
            preferences.edit().putString(KEY_ACCOUNT_BACKDROP, path).apply()
            Toast.makeText(activity, R.string.msg_image_applied, Toast.LENGTH_SHORT).show()
            dismiss()
        }

        cancelBtn.setOnClickListener { dismiss() }
    }
}